api/dps/server_integration_test.go

```diff
@@ -0,0 +1,1024 @@
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+
+		// No data is written in the database, so the index should fail to retrieve anything.
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetRegisterValuesRequest{
+			Height: height,
+			Paths:  [][]byte{mocks.GenericBytes},
+		}
+		_, err = server.GetRegisterValues(context.Background(), req)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestIntegrationServer_GetCollection(t *testing.T) {
+	collections := mocks.GenericCollections(4)
+	collID := collections[0].ID()
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+		writer := index.NewWriter(db, storage)
+
+		// Insert mock data in database.
+		require.NoError(t, writer.Collections(mocks.GenericHeight, collections))
+		require.NoError(t, writer.Close())
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetCollectionRequest{
+			CollectionID: collID[:],
+		}
+		resp, err := server.GetCollection(context.Background(), req)
+
+		require.NoError(t, err)
+		assert.Equal(t, collID[:], resp.CollectionID)
+
+		wantData, err := codec.Marshal(collections[0])
+		require.NoError(t, err)
+		assert.Equal(t, wantData, resp.Data)
+	})
+
+	t.Run("handles indexer failure on Collection", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+
+		// No data is written in the database, so the index should fail to retrieve anything.
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetCollectionRequest{
+			CollectionID: collID[:],
+		}
+		_, err = server.GetCollection(context.Background(), req)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestIntegrationServer_ListCollectionsForHeight(t *testing.T) {
+	collections := mocks.GenericCollections(4)
+	height := mocks.GenericHeight
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+		writer := index.NewWriter(db, storage)
+
+		// Insert mock data in database.
+		require.NoError(t, writer.Collections(height, collections))
+		require.NoError(t, writer.Close())
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.ListCollectionsForHeightRequest{
+			Height: mocks.GenericHeight,
+		}
+		resp, err := server.ListCollectionsForHeight(context.Background(), req)
+
+		require.NoError(t, err)
+		assert.Equal(t, mocks.GenericHeight, resp.Height)
+
+		for _, collection := range collections {
+			wantID := collection.ID()
+			assert.Contains(t, resp.CollectionIDs, wantID[:])
+		}
```

Comment:
We only check that all collections are in the collection IDs. We don't check that there are no duplicates and no extra collections IDs. So I guess we want to add `assert.Len(t, resp.CollectionIDs, len(collections))`.